joern-cli/frontends/php2cpg/src/test/scala/io/joern/php2cpg/querying/MethodTests.scala

```diff
@@ -245,4 +245,67 @@ class MethodTests extends PhpCode2CpgFixture {
       }
     }
   }
+
+  "duplicate function definitions" in {
+    val cpg = code("""<?php
+        |function baz() {}
+        |class Foo {
+        |   function __construct() {
+        |     if (true) {
+        |       function foo() {
+        |         if (true) {
+        |           function bar() {}
+        |         } else if (false) {
+        |           function bar() {}
+        |         } else {
+        |           function bar() {}
+        |         }
+        |       }
+        |     } else {
+        |       function foo() {
+        |         if (true) {
+        |           function bar() {}
+        |         } else if (false) {
+        |           function bar() {}
+        |         } else {
+        |           function bar() {}
+        |         }
+        |       }
+        |     }
+        |   }
+        |}
+        |""".stripMargin)
+
+    inside(cpg.method.name("foo.*").l) {
+      case foo :: duplicateFoo :: Nil =>
+        foo.name shouldBe "foo"
+        foo.fullName shouldBe "Foo.__construct.foo"
+
+        duplicateFoo.name shouldBe "foo<duplicate>0"
+        duplicateFoo.fullName shouldBe "Foo.__construct.foo<duplicate>0"
+      case xs => fail(s"Expected two functions for `foo`, got ${xs.name.mkString("[", ",", "]")}")
+    }
+
+    inside(cpg.method.name("bar.*").l) {
+      case barOne :: barDedupOne :: barDedupTwoOne :: barTwo :: barDedupTwo :: barDedupTwoTwo :: Nil =>
+        barOne.name shouldBe "bar"
+        barOne.fullName shouldBe "Foo.__construct.foo.bar"
+
+        barDedupOne.name shouldBe "bar<duplicate>0"
```

Comment:
For the name of methods and classes please do not add the `<duplicated>` suffix.